components/viaduct/android/src/main/java/mozilla/appservices/httpconfig/HttpConfig.kt

```diff
@@ -4,140 +4,108 @@
 
 package mozilla.appservices.httpconfig
 
-import com.google.protobuf.ByteString
-import mozilla.appservices.support.native.RustBuffer
+import mozilla.appservices.httpconfig.uniffi.FfiBackend
+import mozilla.appservices.httpconfig.uniffi.FfiRequest
+import mozilla.appservices.httpconfig.uniffi.FfiResponse
+import mozilla.appservices.httpconfig.uniffi.Fetcher
+import mozilla.appservices.httpconfig.uniffi.Method
 import mozilla.components.concept.fetch.Client
 import mozilla.components.concept.fetch.MutableHeaders
 import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
 import java.util.concurrent.TimeUnit
-import java.util.concurrent.locks.ReentrantReadWriteLock
-import kotlin.concurrent.read
-import kotlin.concurrent.write
+
+internal fun Method.into(): Request.Method {
+    return when (this) {
+        Method.GET -> Request.Method.GET
+        Method.POST -> Request.Method.POST
+        Method.HEAD -> Request.Method.HEAD
+        Method.OPTIONS -> Request.Method.OPTIONS
+        Method.DELETE -> Request.Method.DELETE
+        Method.PUT -> Request.Method.PUT
+        Method.TRACE -> Request.Method.TRACE
+        Method.CONNECT -> Request.Method.CONNECT
+    }
+}
+
+internal fun FfiRequest.into(): Request {
+    val headers = MutableHeaders()
+    for (h in this.headers) {
+        headers.append(h.key, h.value)
+    }
+    return Request(
+        url = this.url,
+        method = this.method.into(),
+        headers = headers,
+        connectTimeout = Pair(this.connectTimeoutSecs.toLong(), TimeUnit.SECONDS),
+        readTimeout = Pair(this.readTimeoutSecs.toLong(), TimeUnit.SECONDS),
+        body = if (this.body != null) {
+            Request.Body(this.body!!.toByteArray().inputStream())
```

Comment:
is the `!!` necessary here? I've seen it omitted in these scenarios where you explicitly check against null.

Reply:
Yeah I'm not sure 😅 The compiler didn't let me have it with the null check so i added it, what Ben is saying could be true - although I don't expect the conversion of the **same** response  to happen on multiple threads, maybe there is some type of guarantee I can add, I'll explore 🤔